### PR TITLE
Introduce AndroidGradleCacheItemCollector

### PR DESCRIPTION
**Pull request by godrei**
Introduce `AndroidGradleCacheItemCollector` for android gradle cache include and exclude path listing.
This change exposes the default cache item collection logic, without committing the include and exclude paths.
Use-case: https://github.com/bitrise-steplib/steps-fastlane/pull/78/files#diff-b2eb74c816d6381ef0c5306d99c2ba6de8fc5538f5dc57535d7fd7dc36153042.

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -27,34 +27,61 @@ const (
 	LevelAll  = Level("all")
 )
 
-// Collect walks the directory tree underneath projectRoot and registers matching
-// paths for caching based on the value of cacheLevel. Returns an error if there
-// was an underlying error that would lead to a corrupted cache file, otherwise
-// the given path is skipped.
-func Collect(projectRoot string, cacheLevel Level) error {
+// ItemCollector ...
+type ItemCollector interface {
```
Comment (lpusok) [inline cache/cache.go:31]: NOTE: Long term it would be better to have a generic cache interface in go-steputils.

File: cache/cache.go [hunk 2]
```diff
@@ -27,34 +27,61 @@ const (
...
+	CollectCacheItems(dir string, cacheLevel Level) ([]string, []string, error)
```
Comment (lpusok) [inline cache/cache.go:32]: NOTE: The package name is 'cache' so the cache in CollectCacheItems is maybe redundant.

**Review by lpusok — APPROVED** [review]

File: cache/cache.go (hunk 1 above)
Comment (godrei) [reply cache/cache.go]: Moved CacheLevel and ItemCollecto to go-steputils/cache: https://github.com/bitrise-io/go-steputils/pull/31

File: cache/cache.go [hunk 3]
```diff
@@ -14,55 +14,64 @@ import (
...
+	if err != nil {
+		return err
+	}
+
+	if len(includes) == 0 && len(excludes) == 0 {
+		return nil
+	}
+
+	gradleCache := cache.New()
+	gradleCache.IncludePath(strings.Join(includes, "\n"))
```
Comment (lpusok) [inline cache/cache.go:65]: Can this work?
```suggestion
	gradleCache.IncludePath(includes...)
```

File: cache/cache.go [hunk 4]
```diff
@@ -14,55 +14,64 @@ import (
...
+	gradleCache.ExcludePath(strings.Join(excludes, "\n"))
```
Comment (lpusok) [inline cache/cache.go:66]:
```suggestion
	gradleCache.ExcludePath(excludes...)
```

**Review by lpusok — APPROVED** [review]

**Pull request merged**